app/models/energy/constructionset.py

```diff
@@ -0,0 +1,146 @@
+"""Construction Set Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+from app.models.energy.construction import EnergyConstructionOpaque, EnergyConstructionWindow
+
+
+class WallSet(BaseModel):
+    """A set of constructions for wall assemblies."""
+
+    type: Enum('WallSet', {
+               'type': 'WallSet'})
+
+
+    interior_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    ground_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+
+class FloorSet(BaseModel):
+    """A set of constructions for floor assemblies."""
+
+    type: Enum('FloorSet', {
+               'type': 'FloorSet'})
+
+
+    interior_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    ground_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+class RoofCeilingSet(BaseModel): 
+    """A set of constructions for roof and ceiling assemblies."""
+
+    type: Enum('RoofCeilingSet', {
+               'type': 'RoofCeilingSet'})
+
+    interior_ceiling: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_roof: EnergyConstructionOpaque = Schema(
+        default=None
+    )    
+
+    ground_ceiling: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+class ApertureSet(BaseModel): 
+    """A set of constructions for aperture assemblies."""
+
+    type: Enum('ApertureSet', {
+               'type': 'ApertureSet'})
+
+    interior_window: EnergyConstructionWindow = Schema(
+        default=None
+    )
+
+    fixed_window: EnergyConstructionWindow = Schema(
+        default=None
+    )    
+
+    skylight: EnergyConstructionWindow = Schema(
+        default=None
+    )
+
+    operable_window: EnergyConstructionWindow = Schema(
+        default=None
+    ) 
+
+    glass_door: EnergyConstructionWindow = Schema(
+        default=None
+    )
+
+class DoorSet(BaseModel): 
+    """A set of constructions for door assemblies."""
+
+    type: Enum('DoorSet', {
+               'type': 'DoorSet'})
+
+
+    interior_door: EnergyConstructionOpaque = Schema(
```

Comment:
For all of the constructions in the `DoorSet`, I think we can exclude the use of the word `door` from the property names.  So it would just become `interior_construction`, `exterior_construction`, and `overhead_construction`.